akka-http2-support/src/main/scala/akka/http/scaladsl/Http2.scala

```diff
@@ -67,6 +67,7 @@ final class Http2Ext(private val config: Config)(implicit val system: ActorSyste
             httpPlusSwitching(http1, http2).addAttributes(Http.prepareAttributes(settings, incoming))
               .watchTermination()(Keep.right)
               .join(incoming.flow)
+              .addAttributes(Http.cancellationStrategyAttributeForDelay(settings.streamCancellationDelay))
```

Comment:
Could it make sense to only add attributes in one place, combining `Http.prepareAttributes` with the stream cancelation delay one and only do a single `addAttributes`?

Reply:
If we would have a solution for the needed duplication above where we want to enable it for all cuts of the stack that we provide APIs for, then maybe yes. When we cannot solve that, we probably cannot clean it up easily.

Reply:
I think I didn't think that all the way through, we wouldn't want the other `Http.prepareAttributes` to apply to the incoming, so NVM

Reply:
I think I didn't think that all the way through, we wouldn't want the other `Http.prepareAttributes` to apply to the incoming, so NVM